src/cars/car.service.ts

```diff
@@ -1,82 +1,79 @@
-        { search: `%${search}%` },
-      );
-    }
-    try {
-      const cars = await query.getMany();
-      return cars;
-    } catch (error) {
-      throw new InternalServerErrorException();
-    }
+  getCars(userId: number) {
+    return this.prisma.car.findMany({
+      where: {
+        userId,
+      },
+    });
   }
 
-  async getByNumberPlate(numberPlate: string, user: User): Promise<Car> {
-    const car = await this.carsRepository.findOne({
-      where: { numberPlate, user },
+  getCarByNumberPlate(userId: number, numberPlate: string) {
+    return this.prisma.car.findFirst({
+      where: {
+        numberPlate: numberPlate,
+        userId,
+      },
     });
-    if (!car) {
-      throw new NotFoundException(
-        `Car with number plate ${numberPlate} not found.`,
-      );
-    }
-    return car;
   }
 
-  async create(createCarDto: CreateCarDto, user: User): Promise<void> {
-    const car = this.carsRepository.create({ ...createCarDto, user });
-    const { numberPlate } = createCarDto;
-    try {
-      await this.carsRepository.save(car);
-    } catch (error) {
-      if (error.code === ERROR_DUPLICATE_KEY_VALUE) {
-        throw new ConflictException(
-          `The number plate: ${numberPlate} already exists.`,
-        );
-      } else {
-        throw new InternalServerErrorException();
-      }
-    }
+  async createCar(userId: number, carDto: CarDto) {
+    const car = await this.prisma.car.create({
+      data: {
+        userId,
+        ...carDto,
+      },
+    });
+
+    return car;
   }
 
-  async updateByNumberPlate(
+  async editCarByNumberPlate(
+    userId: number,
     numberPlate: string,
-    createCarDto: CreateCarDto,
-    user: User,
-  ): Promise<Car> {
-    const car = await this.getByNumberPlate(numberPlate, user);
-    const changedCar = this.carsRepository.save({
-      ...car,
-      ...createCarDto,
+    carDto: CarDto,
+  ) {
+    const car = await this.prisma.car.findUnique({
+      where: {
+        numberPlate: numberPlate,
+      },
+    });
+
+    // check if user owns the car
+    if (!car || car.userId !== userId)
+      throw new ForbiddenException('Access to resources denied');
+
+    return this.prisma.car.update({
+      where: {
+        numberPlate: numberPlate,
+      },
+      data: {
+        ...carDto,
+      },
     });
-    return changedCar;
   }
 
-  async deleteByNumberPlate(numberPlate: string, user: User): Promise<void> {
-    const car = await this.getByNumberPlate(numberPlate, user);
-    await this.carsRepository.delete(car);
+  async deleteCarByNumberPlate(userId: number, numberPlate: string) {
+    const car = await this.prisma.car.findUnique({
+      where: {
+        numberPlate: numberPlate,
+      },
+    });
+
+    // check if user owns the car
+    if (!car || car.userId !== userId)
```

Comment:
This can be an static `assertCarOwnership` method for throwing exception on this repetitive condition.

Reply:
Maybe I will apply smth global like: `UseGuards` to limit access to certain http calls?